control-plane-operator/controllers/hostedcontrolplane/cloud/openstack/providerconfig.go

```diff
@@ -48,20 +49,23 @@ func ReconcileCloudConfigConfigMap(externalNetworkID *string, cm *corev1.ConfigM
 }
 
 // getCloudConfig returns the cloud config.
-func getCloudConfig(externalNetworkID *string, cloudName string, credentialsSecret *corev1.Secret, caCertData []byte) string {
+func getCloudConfig(platformSpec *hyperv1.OpenStackPlatformSpec, credentialsSecret *corev1.Secret, caCertData []byte) string {
 	config := string(credentialsSecret.Data[CredentialsFile])
 	config += "[Global]\n"
 	config += "use-clouds = true\n"
 	config += "clouds-file=" + CloudCredentialsDir + "/" + CloudsSecretKey + "\n"
-	config += "cloud=" + cloudName + "\n"
+	config += "cloud=" + platformSpec.IdentityRef.CloudName + "\n"
 	// This takes priority over the 'cacert' value in 'clouds.yaml' and we therefore
 	// unset then when creating the initial secret.
 	if caCertData != nil {
 		config += "ca-file=" + CaDir + "/" + CABundleKey + "\n"
 	}
 	config += "\n[LoadBalancer]\nmax-shared-lb = 1\nmanage-security-groups = true\n"
-	if externalNetworkID != nil && *externalNetworkID != "" {
-		config += "floating-network-id = " + *externalNetworkID + "\n"
+	if platformSpec.ExternalNetwork != nil {
+		externalNetworkID := ptr.Deref(platformSpec.ExternalNetwork.ID, "")
```

Comment:
This isn't for now, given it's already an issue, but this means we just ignore `ExternalNetwork` if `Filter` is set, right? Odd that we even supported that in the first place if so. I'm guessing it's inherited from CAPO.

Reply:
Right now we don't support anything but the ID indeed. We supported the name for little time and realized this was unsafe in this case.
I think for now it's ok.